pkg/river/internal/value/decode.go

```diff
@@ -339,6 +339,17 @@ func tryCapsuleConvert(from Value, into reflect.Value, intoType Type) (ok bool,
 		}
 	}
 
+	// Last attempt: see if the Go types can be converted.
+	if from.Reflect().CanAddr() && from.Reflect().Addr().CanConvert(into.Type()) {
```

Comment:
The order here is probably important, since it's more ideal to convert the pointer to avoid a copy.